src/test/rpc/TransactionEntry_test.cpp

```diff
@@ -143,29 +144,43 @@ class TransactionEntry_test : public beast::unit_test::suite
     }
 
     void
-    testRequest()
+    testRequest(unsigned apiVersion)
     {
-        testcase("Basic request");
+        testcase("Basic request API version " + std::to_string(apiVersion));
         using namespace test::jtx;
         Env env{*this};
 
-        auto check_tx = [this, &env](
+        auto check_tx = [this, &env, apiVersion](
                             int index,
                             std::string const txhash,
-                            std::string const expected_json = "") {
+                            std::string const expected_json = "",
+                            std::string const close_time_iso = "") {
             // first request using ledger_index to lookup
-            Json::Value const resIndex{[&env, index, &txhash]() {
+            Json::Value const resIndex{[&env, index, &txhash, apiVersion]() {
                 Json::Value params{Json::objectValue};
                 params[jss::ledger_index] = index;
                 params[jss::tx_hash] = txhash;
+                params[jss::api_version] = apiVersion;
                 return env.client().invoke(
                     "transaction_entry", params)[jss::result];
             }()};
 
             if (!BEAST_EXPECTS(resIndex.isMember(jss::tx_json), txhash))
                 return;
 
-            BEAST_EXPECT(resIndex[jss::tx_json][jss::hash] == txhash);
+            if (apiVersion > 1)
+            {
+                BEAST_EXPECT(resIndex[jss::hash] == txhash);
+                BEAST_EXPECT(resIndex[jss::validated] == true);
+                BEAST_EXPECT(!resIndex[jss::tx_json].isMember(jss::Amount));
+
+                if (!close_time_iso.empty())
```

Comment:
The fix that I proposed causes failures in `AccountTxPaging_test`, so I'm not certain if that's the right way to go about it. LMK if you have other ideas.

Reply:
I think that's because  your proposed change has removed `ledger_index` from inside transaction in API version 1. I avoided this mistake in my attempt to improve `getJson`

Reply:
also, re. your suggestions branch, what's the story behind this suggestion in `LedgerMaster::isValidated` ?

```
         // SQL database doesn't match ledger chain
         clearLedger(seq);
+         app_.getInboundLedgers().acquire(
+                       *hash, seq, InboundLedger::Reason::GENERIC);
```

Reply:
I know that you faithfully copied the previous (API version 1) behavior over to the new place, to ensure backwards compatibility.

Since we have detected that there is a problem with a ledger, we have an opportunity to correct it. After erasing the wrong ledger data, we can fetch the correct ledger through the `acquire` call.

Does any other process in `rippled` rectify the broken ledgers? (I believe some components of FLR can solve this problem)